src/lib/y2storage/guided_proposal.rb

```diff
@@ -167,9 +167,45 @@ def space_maker
     end
 
     # Copy of #initial_devicegraph without all the partitions that must be wiped out
-    # according to the settings
+    # according to the settings. Empty partition tables without preferred type are
+    # deleted and created again with the preferred type.
+    #
+    # @return [Y2Storage::Devicegraph]
     def clean_graph
-      @clean_graph ||= space_maker.delete_unwanted_partitions(initial_devicegraph)
+      return @clean_graph if @clean_graph
+
+      new_devicegraph = initial_devicegraph.dup
+
+      # FIXME: This could change partition tables of devices that are not used
+      # for the proposal.
+      fix_empty_partition_tables(new_devicegraph)
```

Comment:
Didn't we discuss that this method should...?

(A) only delete the current partition table without proposing the new one (that's already done in another part)
(B) return the list of the affected devices (sids or names), so the deleted partition tables can be restored at the end

About (B) I believe we agreed to place a comment here for the future like:
```
# TODO: remember the list of affected devices so we can restore their partition tables at
# the end of the process for those devices that were not used, as soon as libstorage-ng
# allows us to copy sub-graphs.
```